### PR TITLE
Go Input Component

### LINKED ISSUE #119: Create a <go-input> component
### What is the feature that you would like to propose?
We need to create a componentized version of a standard input to make it easier to control implementation and features for the input element.

### Describe the solution you'd like
1. Have a label that corresponds to the input
2. Have a input with a type of text
3. Have a place to put a hint text to add additional information around the input
4. Handle its own errors with the addition of an error hint text.
5. Be able to be disabled
6. Support both a light and a dark theme
7. Be set up to work with Angular's [reactive forms](https://angular.io/guide/reactive-forms)

### PR DESCRIPTION
**Pull request by StevenUlmer**
Closes #119 

This adds an input component with a few bindings to allow for the use within reactive forms. It allows you to pass in both hints and errors to be displayed. It puts the responsibility of validation on the parent component allowing the implementor to have full control over the input.

### CODE REVIEW

File: projects/go-lib/src/lib/components/go-input/go-input.component.html [hunk 1]
```diff
@@ -1,22 +1,27 @@
 <div [formGroup]="parentFormGroup">
-  <label [for]="controlName" class="go-form__label">
+  <label [for]="controlName"
+         class="go-form__label"
+         [ngClass]="{'go-form__label--dark': theme == 'dark'}">
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-input/go-input.component.html]: Should we use `===` here?

File: projects/go-lib/src/lib/components/go-input/go-input.component.html [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
     {{ label }}
   </label>
   <input [id]="controlName"
          class="go-form__input"
-         [ngClass]="{'go-form__input--error': !controlIsValid()}"
+         [ngClass]="{'go-form__input--error': !controlIsValid(), 'go-form__input--dark': theme == 'dark'}"
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-input/go-input.component.html]: Same here, `===`?

File: projects/go-lib/src/lib/components/go-input/go-input.component.html [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
          [type]="inputType"
          [placeholder]="placeholder"
          [formControlName]="controlName">
 
-  <p *ngFor="let hint of hints" class="go-hint">
+  <p *ngFor="let hint of hints" class="go-hint" [ngClass]="{'go-form__input--dark': theme == 'dark'}">
     {{ hint }}
   </p>
 
   <ng-container *ngIf="!controlIsValid()">
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-input/go-input.component.html]: Should we apply this `ngIf` to the div below it to cut out an unnecessary html element?

File: projects/go-lib/src/lib/components/go-input/go-input.component.ts [hunk 4]
```diff
@@ -7,14 +7,16 @@ import { FormGroup } from '@angular/forms';
...
   @Input() hints: string[];
-  @Input() controlName: string;
+  @Input() inputDisabled: boolean;
+  @Input() inputType: string = 'text';
+  @Input() label: string;
   @Input() parentFormGroup: FormGroup;
   @Input() placeholder: string = '';
+  @Input() theme: string = 'light';
 
   constructor() { }
```
Comment (grahamhency) [inline projects/go-lib/src/lib/components/go-input/go-input.component.ts]: Should we throw some errors in the `constructor` if any of the required items are not passed? Just a thought here, but we could do something like:
```
constructor() {
  if (!this.controlName) { throw new Error('GoInput: controlName is a required Input'); }
}
```

Comment (StevenUlmer) [reply projects/go-lib/src/lib/components/go-input/go-input.component.ts]: I could have done it wrong, but when I put this in the constructor it was throwing errors even when passing in the control name as it hadn't been defined yet. I went ahead and added it to the onInit tho

Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-input/go-input.component.ts]: @StevenUlmer, my bad, I meant `onInit`. Thanks though, I think that'll help a lot.

**Pull request merged**